faw/ci/ci_container_service.py

```diff
@@ -1307,6 +1578,101 @@ async def _get_decisions(self, req):
     return t
 
 
+def start_devmount_watcher(*, config_dir, build_dir, devmounts, faw_docker_id, builder_thread):
+
+    import watchfiles
```

Comment:
I agree with this in concept. It makes sense that these threads are all top-level classes. They are the way they are because they follow an original code structure that is no longer really necessary.

Unfortunately, doing a good job of refactoring the threads out is likely to involve a decent bit of testing of the CI features and the different FAW modes to ensure that we are not breaking anything. I would prefer to do that in a different PR, if that looks ok.

Adding @wwoods in case he has comments.